datatypes/mcs_int128.h

```diff
@@ -203,6 +180,36 @@ class TSInt128
       return s128Value == static_cast<int128_t>(x);
     }
 
+    inline operator double() const
+    {
+      return toDouble();
+    }
+
+    inline long double toDouble() const
```

Comment:
Return type here should be `double` and not `long double`.

Reply:
Good catch. I wonder how did this pass through the compiler.